tests/test_worker.py

```diff
@@ -93,3 +96,36 @@ def test_normal_run(self):
         worker.run()
         self.assertEqual(consumer.call_count, 1)
         self.assertEqual(channel_layer.send.call_count, 0)
+
+
+class WorkerGroupTests(ChannelTestCase):
+
+    def test_graceful_stop(self):
+
+        callback_is_running = threading.Event()
+
+        def callback(channel, message):
+            callback_is_running.set()
+            # emulate some delay to validate graceful stop
+            time.sleep(0.1)
+            callback_is_running.clear()
+
+        consumer = mock.Mock()
+        Channel('test').send({'test': 'test'}, immediately=True)
+        channel_layer = channel_layers[DEFAULT_CHANNEL_LAYER]
+        channel_layer.router.add_route(route('test', consumer))
```

Comment:
This route pollute global state an doesn't cleanup it properly.

Reply:
Fair, but I was searching for smth like `.remove_route` and found nothing, so the only way is to drop a route from `router.root.routing` manually and update channels names like it's done [here](https://github.com/django/channels/blob/master/channels/routing.py#L33-L34), right?

Reply:
Or you can create channel layer instance manually with class constructor and use it.

Reply:
Adressed the issue with https://github.com/django/channels/pull/792/files#diff-3ba621217225120eff2c061666b8043cR103, let me know if you like it or not.